packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -32,24 +32,28 @@ enum Orientation {
 /// exception, unless the `nullOk` argument is set to true, in which case it
 /// returns null.
 ///
-/// MediaQueryData includes two [EdgeInsets] values:
-/// [padding] and [viewInsets]. These
-/// values reflect the configuration of the device and are used by
-/// many top level widgets, like [SafeArea] and the Cupertino and
-/// Material scaffold widgets. The padding value defines areas that
-/// might not be completely visible, like the display "notch" on the
-/// iPhone X. The viewInsets value defines areas that aren't visible at
-/// all, typically because they're obscured by the device's keyboard.
+/// MediaQueryData includes three [EdgeInsets] values:
+/// [padding], [viewPadding], and [viewInsets]. These values reflect the
+/// configuration of the device and are used by many top level widgets, like
+/// [SafeArea] and the Cupertino and Material [Scaffold] widgets. The padding
+/// value defines areas that might not be completely visible, like the display
+/// "notch" on the iPhone X. The viewInsets value defines areas that aren't
+/// visible at all, typically because they're obscured by the device's keyboard.
+/// Similar to viewInsets, viewPadding does not differentiate padding in areas
+/// that may be obscured. For example, by using the viewPadding property,
+/// padding would defer to the iPhone "safe area" regardless of whether a
+/// keyboard is showing.
 ///
-/// The viewInsets and padding values are independent, they're both
+/// The viewInsets, viewPadding, and padding values are independent, they're
```

Comment:
This actually isn't quite true - and wasn't true previously.  Padding is a derived property from viewInsets and viewPadding.  So we should be able to say something like `The viewInsets and viewPadding values are independent ....  The padding value is derived from the viewInsets and viewPadding value ...`